conjure-undertow-processor/src/test/java/com/palantir/conjure/java/undertow/processor/sample/DefaultDecoderService.java

```diff
@@ -32,49 +31,55 @@
+            @Handle.QueryParam("optionalInt") OptionalInt optionalIntParam,
+            @Handle.QueryParam("dateTime") OffsetDateTime dateTimeParam,
+            @Handle.QueryParam("ridSetParam") Set<ResourceIdentifier> ridSetParam,
+            @Handle.QueryParam("optionalSafeLongParam") Optional<SafeLong> optionalSafeLongParam,
+            @Handle.QueryParam("uuidParam") UUID uuidParam,
+            @Handle.QueryParam(value = "decoderParam", decoder = StringCollectionDecoder.class) String decoderParam,
+            @Handle.QueryParam("constructor") Constructor constructor,
+            @Handle.QueryParam("ofFactory") OfFactory ofFactory,
+            @Handle.QueryParam("valueOfFactory") ValueOfFactory valueOfFactory);
 
     @Handle(method = HttpMethod.GET, path = "/headers")
     String headers(
-            @Handle.Header(value = "stringParam") String stringParam,
-            @Handle.Header(value = "booleanParam") Boolean booleanParam,
-            @Handle.Header(value = "stringSetParam") Set<String> stringSetParam,
-            @Handle.Header(value = "stringListParam") List<String> stringListParam,
-            @Handle.Header(value = "optionalStringParam") Optional<String> optionalStringParam,
+            @Handle.Header("stringParam") String stringParam,
+            @Handle.Header("booleanParam") Boolean booleanParam,
+            @Handle.Header("stringSetParam") Set<String> stringSetParam,
+            @Handle.Header("stringListParam") List<String> stringListParam,
+            @Handle.Header("optionalStringParam") Optional<String> optionalStringParam,
+            @Handle.Header("floatBoxed") Float floatBoxed,
+            @Handle.Header("floatUnboxed") float floatUnboxed,
             @Handle.Header(value = "decoderParam", decoder = StringCollectionDecoder.class) String decoderParam,
-            @Handle.Header("floatHeaderBoxed") Float floatHeaderBoxed,
-            @Handle.Header("floatHeaderUnboxed") float floatHeaderUnboxed,
-            @Handle.Header("bigInteger") BigInteger bigInteger);
+            @Handle.Header("constructor") Constructor constructor,
+            @Handle.Header("ofFactory") OfFactory ofFactory,
+            @Handle.Header("valueOfFactory") ValueOfFactory valueOfFactory);
 
     @Handle(
             method = HttpMethod.GET,
-            path = "/pathParam/{stringParam}/{booleanParam}/{decoderParam}/{floatBoxed}/{floatUnboxed}/{bigInt}")
+            path = "/pathParam/{stringParam}/{booleanParam}/{decoderParam}/{floatBoxed}/{floatUnboxed}/{bigInt}"
+                    + "/{constructor}/{ofFactory}/{valueOfFactory}")
     String pathParam(
             @Handle.PathParam String stringParam,
             @Handle.PathParam Boolean booleanParam,
             @Handle.PathParam(decoder = StringDecoder.class) String decoderParam,
             @Handle.PathParam Float floatBoxed,
             @Handle.PathParam float floatUnboxed,
-            @Handle.PathParam BigInteger bigInt);
```

Comment:
Changed this to use custom inner types so it's clear what is being tested here (`valueOf` factory methods).

Also added parameters that use constructors and `of` factory methods to ensure those are tested as well.